web/src/engine/websites/decorators/CoreView.ts

```diff
@@ -0,0 +1,261 @@
+ */
+export async function FetchChaptersSinglePageCSS(this: MangaScraper, manga: Manga, query = queryChapters, extractor = ChapterExtractor): Promise<Chapter[]> {
+    const request = new FetchRequest(new URL(manga.Identifier, this.URI).href);
+    let data = await FetchCSS<HTMLLinkElement>(request, queryChaptersAtomFeed);
+    const uri = new URL(data[0].href, this.URI);
+    uri.searchParams.set('free_only', '0'); // 0: include non-free, 1: exclude non-free
+    data = await FetchCSS(new FetchRequest(uri.href), query);
+    return data.map(element => {
+        const { id, title } = extractor.call(this, element, manga);
+        return new Chapter(this, manga, id, title);
+    });
+}
+
+/**
+ * A class decorator that adds the ability to extract all chapters for a given manga from this website using the given CSS {@link query}.
+ * The chapters are extracted from the composed url based on the `Identifier` of the manga and the `URI` of the website.
+ * @param query - A CSS query to locate the elements from which the chapter identifier and title shall be extracted
+ * @param extractor - An Extractor to get chapter infos
+ */
+export function ChaptersSinglePageCSS(query: string = queryChapters, extractor = ChapterExtractor) {
+    return function DecorateClass<T extends Common.Constructor>(ctor: T, context?: ClassDecoratorContext): T {
+        if (context && context.kind !== 'class') {
+            throw new Error(context.name);
+        }
+        return class extends ctor {
+            public async FetchChapters(this: MangaScraper, manga: Manga): Promise<Chapter[]> {
+                return FetchChaptersSinglePageCSS.call(this, manga, query, extractor);
+            }
+        };
+    };
+}
+
+/**********************************************
+ ******** Page List Extraction Methods ********
+ **********************************************/
+
+/**
+ * An extension method for extracting all pages for the given {@link chapter} using the given CSS {@link query}.
+ * The pages are extracted from the composed url based on the `Identifier` of the {@link chapter} and the `URI` of the website.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param chapter - A reference to the {@link Chapter} which shall be assigned as parent for the extracted pages
+ * @param query - A CSS query to locate the JSON from which the page information shall be extracted
+ */
+async function FetchPagesSinglePageJSON(this: MangaScraper, chapter: Chapter, query = queryEpisodeJSON): Promise<Page[]> {
+    const request = new FetchRequest(new URL(chapter.Identifier, this.URI).href);
+    const dataElement = await FetchCSS(request, query);
+    const data: ChapterJSON = JSON.parse(dataElement[0].dataset.value);
+    if (!data.readableProduct.isPublic && !data.readableProduct.hasPurchased) {
+        throw new Error(`The chapter '${chapter.Title}' is neither public, nor purchased!`);
```

Comment:
Text that may be presented to the user should be [localized](https://github.com/manga-download/haruneko/blob/master/web/src/engine/websites/decorators/Common.ts#L157).

Reply:
Sure. Should i push the update on another branch/PR? Cause i need to modify unrelated files for that